rust/numaflow-core/src/pipeline.rs

```diff
@@ -179,12 +181,16 @@ async fn start_source_forwarder(
     )
     .await?;
 
-    let pending_reader = shared::metrics::create_pending_reader(
-        &config.metrics_config,
-        LagReader::Source(source.clone()),
-    )
-    .await;
-    let _pending_reader_handle = pending_reader.start(is_mono_vertex()).await;
+    let mut _pending_reader_handle: Option<PendingReaderTasks> = None;
+    // only check the pending and lag for source for pod_id = 0
```

Comment:
No need for `mut` here, can just do it inline with an `if`, same in MonoVertex code.